generator/FactoryMethod.php

```diff
@@ -214,6 +214,21 @@ public function getFullName()
         return $this->getClassName() . '::' . $this->getName();
     }
 
+    public function getReturnType(): ?string
```

Comment:
Looks good.

Have you checked that functions generated this way actually work (I'm not sure we that correctness of the generated file)?

Reply:
yes, this is tested with phpstan, too. But I verified it manually as well

Reply:
> code style fixes

Why? Any PR code won't automatically code style fixed prior to merging. That's why I don't see any benefits doing this as part of the build.

> then compile

It's not beneficial to perform compilation before running PHPStan/ PHPUnit because it should be done by whoever sends the PR. 

However, adding a separate GitHub Actions job that will do this could be helpful:

1. compile code
2. fail the build when the `Hamcrest.php` file (build during compilation) is changed now

> then run phpstan
> then run the tests

Doing these one after another won't show what unit tests failed, when PHPStan finds something. Yes, they both need to pass, but for visibility purposes we'd better run them in parallel. It's also faster (I hope).

Reply:
> I wasnt talking about the CI. But the order before commiting in local development

I see.

Anyway I belive that adding separate PhpStan and compile jobs is a good idea.